universe/auto_syncer.go

```diff
@@ -257,8 +257,10 @@ func (f *FederationEnvoy) pushProofToFederation(uniID Identifier, key BaseKey,
 	pushNewProof := func(ctx context.Context, addr ServerAddr) error {
 		remoteUniverseServer, err := f.cfg.NewRemoteRegistrar(addr)
 		if err != nil {
-			return fmt.Errorf("unable to connect to remote "+
-				"server(%v): %v", addr.HostStr(), err)
+			log.Warnf("cannot push proof unable to connect "+
+				"to remote server(%v): %v", addr.HostStr(),
+				err)
+			return nil
 		}
 
 		_, err = remoteUniverseServer.RegisterIssuance(
```

Comment:
I think that if we encounter and return an error at this point then the `errgroup.Group` of the `fn.ParSlice` function call below will cancel the context for all servers.

We should probably warn rather than return an error here also.

Reply:
I think if you can't register a proof with an universe _after_ being able to connect to that universe, I think that might warrant bubbling up the error. As that could point to something being actually wrong (as opposed to the server just being offline).

Reply:
An attacker might run a universe server that would accept a connection but maliciously returns a failure when attempting to submit proofs to it (perhaps for only certain coins etc). Which would interrupt a user's ability to register a proof with all other servers.

I think we should bubble up an error if we fail in any way for **every** universe server. I don't think `fn.ParSlice` (because of `errgroup.Group`) is the right way to go in the long term. But i think further changes belong in a different PR.